Observer/Customer/GuestSubscriberCreated.php

```diff
@@ -43,21 +43,28 @@ public function __construct(
      */
     public function executeWhenEnabled(\Magento\Framework\Event\Observer $observer)
     {
-        if (! $this->registry->registry(self::REGISTRY_KEY_NEW_GUEST_SUBSCRIBER)) {
-            return;
-        }
-
         $config = $this->configFactory->createForCurrentScope();
 
         $email = $this->request->getParam('email');
 
         $subscriber = $this->subscriberFactory->create()->loadByEmail($email);
         $newSubscriberSubscribed = $subscriber->isSubscribed();
 
-        // We only force subscription status in Drip when subscribed because if
-        // the user already exists in Drip and is subscribed there, we don't
-        // want to unsubscribe them, because presumably they have opted in
-        // elsewhere.
-        $this->customerHelper->proceedGuestSubscriberNew($subscriber, $config, $newSubscriberSubscribed);
+        $customer = $this->customerHelper->getCustomerByEmail($email, $config);
+        if ($customer->getId() !== null) {
```

Comment:
Same comment as M1 about possibly reversing the equality and the order of clauses.